idaes/generic_models/unit_models/tests/test_heat_exchanger_lc.py

```diff
@@ -381,6 +395,55 @@ def test_solve(self, model):
         # Check for optimal solution
         assert check_optimal_termination(results)
 
+    @pytest.mark.unit
+    @pytest.mark.xfail(reason="static flowsheet with dynamic heat balance",
+                       raises=DAE_Error)
+    def test_static_flowsheet(self, static_flowsheet_model):
+        m = static_flowsheet_model
+        m.fs.unit = HeatExchangerLumpedCapacitance(default={
+            "shell": {"property_package": m.fs.properties},
+            "tube": {"property_package": m.fs.properties},
+            "dynamic": False,
+            "dynamic_heat_balance": True})

```

Comment:
For these cases, we generally prefer the following to an `xfail`:

```
with pytest.raises(DAE_Error, match="error message"):
    m.fs.unit = HeatExchangerLumpedCapacitance(default={
            "shell": {"property_package": m.fs.properties},
            "tube": {"property_package": m.fs.properties},
            "dynamic": False,
            "dynamic_heat_balance": True})
```

The reason being that this is actually a successfully test of an expected failure, so this way the test shows up as passed, plus it also checks that the error message is what was expected.

Reply:
That makes sense. I didn't do this for the InconsistentUnitsError though, since that's not something we really intended. And this doesn't seem to be caused by IDAES. So I'll try and setup a simple example and submit an issue to the Pyomo folks.

Reply:
@rustygentile Pyomo is already aware of the issue with units: https://github.com/Pyomo/pyomo/issues/1790. However, more people commenting about this being a problem might help motivate a fix.